jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/model/ProcessAssetDesc.java

```diff
@@ -210,6 +215,23 @@ public void setSignals(Collection<String> signals) {
         this.signals = signals;
     }
 
+    @Override
+    public Collection<SignalDesc> getSignalsMetadata() {
+        return signalsDesc;
+    }
+
+    public void setSignalsMetadata(Collection<Signal> signals) {
```

Comment:
this should not be here. create a dependency with the parser.
try to put this logic in ServicesProcessDataEventListener